From e36cc7a188fc3625a3ff44ff759d10cb26808d15 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <grinberg@compulab.co.il>
Date: Wed, 6 Mar 2013 14:05:03 +0200
Subject: ARM: i.MX6: cm-fx6: remove unused HDMI code

cm-fx6 does not use HDCP, so remove HDCP related code.

Signed-off-by: Igor Grinberg <grinberg@compulab.co.il>
---
 arch/arm/mach-mx6/board-cm-fx6-dl.h |   10 ----------
 arch/arm/mach-mx6/board-cm-fx6-q.h  |   10 ----------
 arch/arm/mach-mx6/board-cm-fx6.c    |   26 --------------------------
 3 files changed, 46 deletions(-)

diff --git a/arch/arm/mach-mx6/board-cm-fx6-dl.h b/arch/arm/mach-mx6/board-cm-fx6-dl.h
index f92bcd5..afb09cc 100644
--- a/arch/arm/mach-mx6/board-cm-fx6-dl.h
+++ b/arch/arm/mach-mx6/board-cm-fx6-dl.h
@@ -200,13 +200,3 @@ static iomux_v3_cfg_t cm_fx6_dl_gpmi_nand[] = {
 	MX6DL_PAD_SD4_CLK__RAWNAND_WRN,
 };
 #endif
-
-static iomux_v3_cfg_t cm_fx6_dl_hdmi_ddc_pads[] = {
-	MX6DL_PAD_KEY_COL3__HDMI_TX_DDC_SCL, /* HDMI DDC SCL */
-	MX6DL_PAD_KEY_ROW3__HDMI_TX_DDC_SDA, /* HDMI DDC SDA */
-};
-
-static iomux_v3_cfg_t cm_fx6_dl_i2c2_pads[] = {
-	MX6DL_PAD_KEY_COL3__I2C2_SCL,	/* I2C2 SCL */
-	MX6DL_PAD_KEY_ROW3__I2C2_SDA,	/* I2C2 SDA */
-};
diff --git a/arch/arm/mach-mx6/board-cm-fx6-q.h b/arch/arm/mach-mx6/board-cm-fx6-q.h
index 3c06b6b..dd4414a 100644
--- a/arch/arm/mach-mx6/board-cm-fx6-q.h
+++ b/arch/arm/mach-mx6/board-cm-fx6-q.h
@@ -212,13 +212,3 @@ static iomux_v3_cfg_t cm_fx6_q_gpmi_nand[] = {
 	MX6Q_PAD_SD4_CLK__RAWNAND_WRN,
 };
 #endif
-
-static iomux_v3_cfg_t cm_fx6_q_hdmi_ddc_pads[] = {
-	MX6Q_PAD_KEY_COL3__HDMI_TX_DDC_SCL, /* HDMI DDC SCL */
-	MX6Q_PAD_KEY_ROW3__HDMI_TX_DDC_SDA, /* HDMI DDC SDA */
-};
-
-static iomux_v3_cfg_t cm_fx6_q_i2c2_pads[] = {
-	MX6Q_PAD_KEY_COL3__I2C2_SCL,	/* I2C2 SCL */
-	MX6Q_PAD_KEY_ROW3__I2C2_SDA,	/* I2C2 SDA */
-};
diff --git a/arch/arm/mach-mx6/board-cm-fx6.c b/arch/arm/mach-mx6/board-cm-fx6.c
index ac40c14..537bedf 100644
--- a/arch/arm/mach-mx6/board-cm-fx6.c
+++ b/arch/arm/mach-mx6/board-cm-fx6.c
@@ -805,34 +805,8 @@ static void cm_fx6_hdmi_init(int ipu_id, int disp_id)
 		mxc_iomux_set_gpr_register(0, 0, 1, 1);
 }
 
-/*
- * cm-fx6 shares i2c2 with hdmi ddc, the default is i2c2 function,
- * when hdcp is enabled, the pins should work at ddc function.
- */
-static void hdmi_enable_ddc_pin(void)
-{
-	if (cpu_is_mx6dl())
-		mxc_iomux_v3_setup_multiple_pads(cm_fx6_dl_hdmi_ddc_pads,
-			ARRAY_SIZE(cm_fx6_dl_hdmi_ddc_pads));
-	else
-		mxc_iomux_v3_setup_multiple_pads(cm_fx6_q_hdmi_ddc_pads,
-			ARRAY_SIZE(cm_fx6_q_hdmi_ddc_pads));
-}
-
-static void hdmi_disable_ddc_pin(void)
-{
-	if (cpu_is_mx6dl())
-		mxc_iomux_v3_setup_multiple_pads(cm_fx6_dl_i2c2_pads,
-			ARRAY_SIZE(cm_fx6_dl_i2c2_pads));
-	else
-		mxc_iomux_v3_setup_multiple_pads(cm_fx6_q_i2c2_pads,
-			ARRAY_SIZE(cm_fx6_q_i2c2_pads));
-}
-
 static struct fsl_mxc_hdmi_platform_data cm_fx6_hdmi_data = {
 	.init = cm_fx6_hdmi_init,
-	.enable_pins = hdmi_enable_ddc_pin,
-	.disable_pins = hdmi_disable_ddc_pin,
 };
 
 static struct fsl_mxc_hdmi_core_platform_data cm_fx6_hdmi_core_data = {
-- 
1.7.9.5

